src/data_router.py

```diff
@@ -0,0 +1,71 @@
+import glob
+import logging
+import multiprocessing
+import os
+import random
+from flask import json
+
+from config import RasaNLUConfig
+from rasa_nlu.train import do_train
+
+
+class DataRouter(object):
+    def __init__(self, config, interpreter, emulator):
+        self.ID = random.random()
+        self.config = config
+        self.interpreter = interpreter
+        self.emulator = emulator
+        self.logfile = config['write'].replace(".json", "-{}.json".format(os.getpid()))
+        self.responses = self.create_query_logger(self.logfile)
+        self.train_proc = None
+        self.model_dir = config['path']
+        self.token = config['token']
+
+    def create_query_logger(self, path):
+        logger = logging.getLogger('query-logger')
+        logger.setLevel(logging.INFO)
+        ch = logging.FileHandler(path)
+        ch.setFormatter(logging.Formatter('%(message)s'))
+        logger.propagate = False
+        logger.addHandler(ch)
+        return logger
+
+    def extract(self, data):
+        return self.emulator.normalise_request_json(data)
+
+    def parse(self, text):
+        result = self.interpreter.parse(text)
+        self.responses.info(json.dumps(result, sort_keys=True))
+        return result
+
+    def format(self, data):
```

Comment:
maybe call this `_format` or something else? I'm not sure when we'd ever serialise this object but I'd prefer not to have the same name as python built-in funcs